generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FunctionDataFetcher.kt

```diff
@@ -120,12 +120,15 @@ open class FunctionDataFetcher(
         }
 
     /**
-     * Once all parameters values are properly converted, this function will be called to run a simple blocking function.
-     * If you need to override the exception handling you can override this method.
+     * Once all parameter values are properly converted, this function will be called to run a simple blocking function.
+     * If you need to override the exception handling, you can override this method.
      */
-    protected open fun runBlockingFunction(parameterValues: Map<KParameter, Any?>): Any? = try {
-        fn.callBy(parameterValues)
-    } catch (exception: InvocationTargetException) {
-        throw exception.cause ?: exception
-    }
+    protected open fun runBlockingFunction(parameterValues: Map<KParameter, Any?>): CompletableFuture<Any?>  =
+        CompletableFuture<Any?>().apply {
```

Comment:
same as above, this change is going to cause incompatibility with the update to [graphql-java 22](https://github.com/graphql-java/graphql-java/releases/tag/v22.0)